libcontainer/nsenter/nsexec.c

```diff
@@ -356,116 +395,96 @@ void nsexec(void)
 
-	// required clone_flags to be passed
-	if (config.cloneflags == -1) {
-		pr_perror("Missing clone_flags");
-		exit(1);
-	}
-	// prepare sync pipe between parent and child. We need this to let the
-	// child know that the parent has finished setting up
-	if (pipe(syncpipe) != 0) {
-		pr_perror("Failed to setup sync pipe between parent and child");
-		exit(1);
-	}
+	/* clone(2) flags are mandatory. */
+	if (config.cloneflags == -1)
+		bail("missing clone_flags");
+
+	/* Pipe so we can tell the child when we've finished setting up. */
+	if (pipe(syncpipe) < 0)
+		bail("failed to setup sync pipe between parent and child");
+
+	/* Set up the jump point. */
+	if (setjmp(env) == JUMP_VAL) {
+		/*
+		 * We're inside the child now, having jumped from the
+		 * start_child() code after forking in the parent.
+		 */
 
-	if (setjmp(env) == 1) {
-		// Child
 		uint8_t s = 0;
-		int	consolefd = config.consolefd;
+		int consolefd = config.consolefd;
 
-		// close the writing side of pipe
+		/* Close the writing side of pipe. */
 		close(syncpipe[1]);
 
-		// sync with parent
-		if ((read(syncpipe[0], &s, 1) != 1) || (s != 1)) {
-			pr_perror("Failed to read sync byte from parent");
-			exit(1);
-		}
+		/* Sync with parent. */
+		if (read(syncpipe[0], &s, sizeof(s)) != sizeof(s)
+		    || s != SYNC_VAL)
+			bail("failed to read sync byte from parent");
 
-		if (setsid() == -1) {
-			pr_perror("setsid failed");
-			exit(1);
-		}
+		if (setsid() < 0)
+			bail("setsid failed");
 
-		if (setuid(0) == -1) {
-			pr_perror("setuid failed");
-			exit(1);
-		}
+		if (setuid(0) < 0)
+			bail("setuid failed");
 
-		if (setgid(0) == -1) {
-			pr_perror("setgid failed");
-			exit(1);
-		}
-    
-		if (setgroups(0, NULL) == -1) {
-			pr_perror("setgroups failed");
-			exit(1);
-		}
+		if (setgid(0) < 0)
+			bail("setgid failed");
+
+		if (setgroups(0, NULL) < 0)
+			bail("setgroups failed");
 
 		if (consolefd != -1) {
-			if (ioctl(consolefd, TIOCSCTTY, 0) == -1) {
-				pr_perror("ioctl TIOCSCTTY failed");
-				exit(1);
-			}
-			if (dup3(consolefd, STDIN_FILENO, 0) != STDIN_FILENO) {
-				pr_perror("Failed to dup stdin");
-				exit(1);
-			}
-			if (dup3(consolefd, STDOUT_FILENO, 0) != STDOUT_FILENO) {
-				pr_perror("Failed to dup stdout");
-				exit(1);
-			}
-			if (dup3(consolefd, STDERR_FILENO, 0) != STDERR_FILENO) {
-				pr_perror("Failed to dup stderr");
-				exit(1);
-			}
+			if (ioctl(consolefd, TIOCSCTTY, 0) < 0)
+				bail("ioctl TIOCSCTTY failed");
+			if (dup3(consolefd, STDIN_FILENO, 0) != STDIN_FILENO)
+				bail("failed to dup stdin");
+			if (dup3(consolefd, STDOUT_FILENO, 0) != STDOUT_FILENO)
+				bail("failed to dup stdout");
+			if (dup3(consolefd, STDERR_FILENO, 0) != STDERR_FILENO)
+				bail("failed to dup stderr");
 		}
 
-		// Finish executing, let the Go runtime take over.
+		/* Finish executing, let the Go runtime take over. */
 		return;
 	}
 
-	// Parent
+	/* Run the parent code. */
```

Comment:
Note that I ran with `-linux -l120` because the 80 character restriction isn't something we enforce anywhere else (and it makes the code less readable).